Rho.md

```diff
@@ -0,0 +1,119 @@
+
+You can insert header captures and body decoders for your requests, an example:
+```scala
+val api = new RhoService[IO] {
+    "Description of api endpoint" **
+    GET / "somePath" / pathVar[Int]("someInt", "parameter description") >>> auth ^ jsonOf[T] |>> { (someInt: Int, au: AuthInfo, body: T) => {
+            Ok("result")
+        }
+    }
+}
+```
+`auth` captures the authentication information. We are specifying that the body is a json object for some T.
+Anything that has a `EntityDecoder` can be specified as a body decoder. In the example above `jsonOf` is an `EntityDecoder` provided by `circe`. Included decoders in Rho are:
+- binary
+- binaryChunk
+- byteArrayDecoder
+- text
+- charArrayDecoder
+- binFile
+- textFile
+- multipart
+- void
+
+This will all work automatically for the most part and that's all you need. But sometimes the models for apis are too big and complex and you need documentation on every field in your model. This type of information is not captured in Rho right now and we have to provide specifications for those manually. For that we can make use of `withCustomSerializer` in `SwaggerSupport`.
+
+```scala
+case class MyClass(name: String, description: String, someBool: Boolean, tags: List[String], someObj: Map[String, String])
+//or
+case class MyClass(name: String, description: String, someBool: Boolean, tags: List[String], someObj: JsonObject)
+
+val myClassModel: Set[Model] = Set(
+    ModelImpl(
+      id = "MyClass",
+      id2 = "MyClass",
+      `type` = "object".some,
+      description = "MyClass".some,
+      name = "MyClass".some,
+      properties = Map(
+        "name" -> StringProperty(
+          required = true,
+          description = "name of MyClass".some,
+          enums = Set()
+        ),
+        "category" -> StringProperty(
+          required = true,
+          description = "enum of category".some,
+          enums = Set("A", "B", "C")
+        ),
+        "someBool" -> AbstractProperty(
+          required = true,
+          description =
+            "boolean".some,
+          format = "bool".some
+        ),
+        "tags" -> ArrayProperty(
+          required = true,
+          description = "some tags".some,
+          items = StringProperty(required = true,
+                                 enums = Set(),
+                                 description = "tag name".some),
+          uniqueItems = true
+        ),
+        "someObj" -> ObjectProperty(
+          required = true,
+          description = "some object".some
+        )
+      ),
+      example =
+        """{"name" : "MyClass1",  "category" : "A", "tags" : ["tag1", "tag2"], "someBool" : false, "someObj": {}}""".some
+    )
+  )
+
+// register this model
+SwaggerSupport[IO].createRhoMiddleware(
+              swaggerFormats = DefaultSwaggerFormats
```

Comment:
I'm confused with this indentation. Maybe 104 and 105 should be in a single line?

Reply:
Do you still feel that with the changes, this was scalafmt btw.

Reply:
Yes, but if it was scalafmt, maybe it is not worth fixing :(